homeassistant/components/remote/__init__.py

```diff
@@ -0,0 +1,151 @@
+"""
+Component to interface with universal remote control devices.
+
+For more details about this component, please refer to the documentation
+at https://home-assistant.io/components/remote/
+"""
+from datetime import timedelta
+import logging
+import os
+
+import voluptuous as vol
+from homeassistant.config import load_yaml_config_file
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.entity import ToggleEntity
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (
+    STATE_ON, SERVICE_TURN_ON, SERVICE_TURN_OFF, ATTR_ENTITY_ID)
+from homeassistant.components import group
+from homeassistant.helpers.config_validation import PLATFORM_SCHEMA  # noqa
+
+ATTR_DEVICE = 'device'
+ATTR_COMMAND = 'command'
+ATTR_ACTIVITY = 'activity'
+SERVICE_SEND_COMMAND = 'send_command'
+SERVICE_SYNC = 'sync'
+ATTR_DEFAULT = ''
+
+DOMAIN = 'remote'
+SCAN_INTERVAL = 30
+
+GROUP_NAME_ALL_REMOTES = 'all remotes'
+ENTITY_ID_ALL_REMOTES = group.ENTITY_ID_FORMAT.format('all_remotes')
+
+ENTITY_ID_FORMAT = DOMAIN + '.{}'
+
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
+
+REMOTE_SERVICE_SCHEMA = vol.Schema({
+    vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+    vol.Optional(ATTR_DEVICE): cv.string,
+    vol.Optional(ATTR_COMMAND): cv.string,
+    vol.Optional(ATTR_ACTIVITY): cv.string
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def is_on(hass, entity_id=None):
+    """Return if the remote is on based on the statemachine."""
+    entity_id = entity_id or ENTITY_ID_ALL_REMOTES
+    return hass.states.is_state(entity_id, STATE_ON)
+
+
+def turn_on(hass, activity=None, entity_id=None):
```

Comment:
the config block lets you specify a default activity to use when turn_on is called without an activity ID